flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -177,11 +179,27 @@ public IncrementalRocksDBSnapshotResources syncPrepareResources(long checkpointI
             return registry -> SnapshotResult.empty();
         }
 
+        final Set<StateHandleID> baseSstFiles;
+        final CheckpointType.SharingFilesStrategy sharingFilesStrategy =
+                checkpointOptions.getCheckpointType().getSharingFilesStrategy();
+        switch (sharingFilesStrategy) {
+            case FORWARD_BACKWARD:
+                baseSstFiles = snapshotResources.baseSstFiles;
+                break;
+            case FORWARD:
+            case NO_SHARING:
```

Comment:
Does this code support `NO_SHARING` mode at the moment? I think newly created files will be used in future snapshots?

Reply:
The `NO_SHARING` mode is used for `SAVEPOINTS`, so we do need to support it in general. This mislead me a bit and made me think we need to support it here. However, savepoints should never end up in this strategy, but should use `RocksFullSnapshotStrategy`. 

Good point! We could throw an exception for `NO_SHARING` as well for now until we properly support incremental savepoints. Sorry for not being careful here :(